### PR TITLE
Remove legacy code for older versions of Keycloak JS

### LINKED ISSUE #500: Remove legacy detection code for old Keycloak JS
#499 introduced some code to ensure Keycloak JS remains compatible. After the 26 release this code can be removed, and we can safely assume the latest.

Probably also a cleanup is in order.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #500

### CODE REVIEW

**Review by stianst — APPROVED** [review]
This is too complex and adds a lot of stuff irrelevant to the website in general to common stuff.

It loads `keycloak.js` for all pages, and not just the app, which is just wasteful, and dumps the whole keycloak package into what is served by the website and not just the single file that is needed.

A much better approach would be to isolate all of this within the AppBuilder. Move `pages/app/index.ftl` to `templates/app.ftl` and run FreeMarker separately within AppBuilder to build `target/web/app/index.html` out of it, as well as copying the single `keycloak.js` file that is needed to `target/web/app/keycloak.js` then you can simply just hardcode the link to that file rather than the ImportMap stuff.

**Review by stianst — CHANGES_REQUESTED** [review]
Was supposed to request changes, and not approve :/

**Comment by jonkoops** [comment]
> It loads `keycloak.js` for all pages, and not just the app, which is just wasteful, and dumps the whole keycloak package into what is served by the website and not just the single file that is needed.

It does not load it for all pages, the inclusion of an import map does not mean the code will be loaded. It is only a piece of metadata that maps module specifiers (in this case `keycloak-js`) to a file path. Only when this specifier is matched on an import will this file be loaded (such as in `app.js`).

There is a valid reason why the whole package is copied in and that is if we decide to modularize the Keycloak JS code in the future we need to be able to load other files as well. For example, `keycloak.js` might try to import some code from `utils.js` from the package. If this file is not present then this will result in an error on the module graph, and the code will not execute. There are essentially three options here:

- Bundle all the code together into a single file using a bundler (introduces another tool).
- Analyze the code and resolve the module graph with all dependencies and copy only required files in the graph (would require analyzing JS code)
- Copy all the files that might be needed and load them when imported (what we do now)

If the overhead is too much I can reduce the contents being copied here to just `.js` files, since the other files cannot be imported anyways.

> A much better approach would be to isolate all of this within the AppBuilder. Move `pages/app/index.ftl` to `templates/app.ftl` and run FreeMarker separately within AppBuilder to build `target/web/app/index.html` out of it, as well as copying the single keycloak.js file that is needed to `target/web/app/keycloak.js` then you can simply just hardcode the link to that file rather than the ImportMap stuff.

I want to keep the import map around as it will make it much easier to future proof since it will be able to handle any NPM dependencies possibly added to Keycloak JS. For example, if we were to hypothetically introduce a dependency called `foo` we can add it to the import map and would simply resolve without the need to modify the Keycloak JS source code.

The import map must appear in the `<head>` before any other `<script>` tags, it cannot be placed anywhere else. If we can build a full HTML page out of it in the `AppBuilder` we'll have to pass in the import map to the parent template. I think this is an option, and it would prevent the import map from showing up in other pages (even though it will never create additional requests!).

Note that there are tooling options here that are more mature than my custom implementation here that we might want to consider. Dealing with the NPM ecosystem in Maven could be made easier by using [`mvnpm`](https://mvnpm.org/), but I have yet to play around with it.

**Comment by jonkoops** [comment]
After some discussion with @stianst I've made the following changes:

- Only `.js` files are now copied from the package, since we cannot use any of the other anyway.
- Instead of writing to an intermediate JSON file, the `AppBuilder` is now responsible for rendering the template for said page.

**Review by douglaspalmer — APPROVED** [review]
LGTM

File: src/main/java/org/keycloak/webbuilder/builders/AppBuilder.java [hunk 1]
```diff
@@ -3,49 +3,94 @@
...
+        if (dotIndex >= 0) {
+            return filename.substring(dotIndex + 1);
         }
+        return "";
     }
 
     @Override
     protected String getTitle() {
-        return "App";
+        return "Vendor";
```
Comment (stianst) [inline src/main/java/org/keycloak/webbuilder/builders/AppBuilder.java:94]: Vendor makes no sense - this is rendering `app/index.html` so should be `App` or perhaps `Mock App` or whatever.

```suggestion
        return "App";
```

**Review by stianst — APPROVED** [review]

File: src/main/java/org/keycloak/webbuilder/builders/AppBuilder.java (hunk 1 above)
Comment (jonkoops) [reply src/main/java/org/keycloak/webbuilder/builders/AppBuilder.java]: Ah yes, forgot to rename this back, thanks.

**Pull request merged**